src/frontend/src/handler/query.rs

```diff
@@ -80,9 +88,23 @@ pub async fn handle_query(
 
     let rows_count = match stmt_type {
         StatementType::SELECT => rows.len() as i32,
+        StatementType::INSERT | StatementType::DELETE | StatementType::UPDATE => {
+            let first_row = rows[0].values();
+            let affected_rows_str = first_row[0]
+                .as_ref()
+                .expect("compute node should return affected rows in output");
+            String::from_utf8(affected_rows_str.to_vec())
+                .unwrap()
+                .parse()
+                .unwrap_or_default()
+        }
         _ => unreachable!(),
     };
 
+    // Implicitly flush the writes.
+    if session.config().get_implicit_flush() {
+        dml::flush_for_write(&session, stmt_type).await?;
```

Comment:
Can we also move this function out of `dml.rs`?

Reply:
I was thinking. But indeed flush_for_write is something relates to dml..

Reply:
Fixed